### PR TITLE
cluster builder should use a job not a pod #349

### LINKED ISSUE #349: cluster builder should use a job not a pod
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Cluster builder is currently creating a pod
https://github.com/kubeflow/fairing/blob/be8d42957077cb9844fcaed0f8cada8c74a982fc/fairing/builders/cluster/cluster.py#L72

I think it would be better to create a K8s Job. 

i also don't think we need to explicitly delete the job after its done. We can rely on the K8s TTL that's being added for jobs.

Having the job spec is useful for debugging.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by nrchakradhar**
**What this PR does / why we need it**:
Changes cluster builder to use Job instead of Pod so that debugging is easier.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #349 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.
No
**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/379)

### CODE REVIEW

**Comment by jinchihe** [comment]
/ok-to-test

@nrchakradhar Thanks for the contribution!

**Comment by nrchakradhar** [comment]
@jinchihe
I see that there is pylint error in the log but do not see any details as to what is the error. Can you please help me how to find the exact error. Thanks

**Comment by jinchihe** [comment]
@nrchakradhar 
```
INFO|2019-09-18T00:55:35|util.py:71| ************* Module kubeflow.fairing.builders.cluster.cluster
INFO|2019-09-18T00:55:35|util.py:71| C: 72, 0: Trailing whitespace (trailing-whitespace)
```

Remove useless whitespace in line 72.

**Comment by nrchakradhar** [comment]
> @nrchakradhar
> 
> ```
> INFO|2019-09-18T00:55:35|util.py:71| ************* Module kubeflow.fairing.builders.cluster.cluster
> INFO|2019-09-18T00:55:35|util.py:71| C: 72, 0: Trailing whitespace (trailing-whitespace)
> ```
> 
> Remove useless whitespace in line 72.

Thanks. I was not able to understand the output. I will submit the change in sometime.

**Comment by nrchakradhar** [comment]
@jinchihe 
Not sure why Travis build is failing. Looks like a downloading error.

> Reading https://pypi.org/simple/azure-mgmt-recoveryservicesbackup/
Downloading https://files.pythonhosted.org/packages/c5/af/13b10d76bc0ff9188c9e5cd9f2726a5745029b76125938491f9abac5399d/azure_mgmt_recoveryservicesbackup-0.3.0-py2.py3-none-any.whl#sha256=5c44bd73df6eb55382335f16a21aee62ddc8e271cd913b9147768f42aba59c3d
error: [Errno 104] Connection reset by peer
The command "python setup.py install" failed and exited with 1 during .

File: kubeflow/fairing/builders/cluster/cluster.py [hunk 1]
```diff
@@ -80,18 +79,31 @@ def build(self):
...
-            namespace=created_pod.metadata.namespace,
+            name=created_job.metadata.name,
+            namespace=created_job.metadata.namespace,
             selectors=labels,
             container="kaniko")
 
-        # clean up created pod and secret
+        # Invoke upstream clean ups
         self.context_source.cleanup()
-        client.CoreV1Api().delete_namespaced_pod(
```
Comment (jinchihe) [inline kubeflow/fairing/builders/cluster/cluster.py]: I suggest add one option to let user choose is delete or not, by default, do not delete. for example, add a option `cleanup=False`, only user set to `True`, will delete resource, what do you think?

**Comment by jinchihe** [comment]
@nrchakradhar Thanks for the great contribution!
/retest

**Comment by jinchihe** [comment]
Thanks @nrchakradhar 
/retest
/lgtm

File: kubeflow/fairing/builders/cluster/cluster.py (hunk 1 above)
Comment (nrchakradhar) [reply kubeflow/fairing/builders/cluster/cluster.py]: Included cleanup flag.

**Pull request merged**